python/paddle/fluid/tests/unittests/collective/process_group_gloo.py

```diff
@@ -204,6 +204,30 @@ def test_create_process_group_gloo(self):
             assert np.array_equal(tensor_y, out2)
         print("test scatter api ok\n")
 
+        # test Gather
+        def test_gather(root):
+            in_shape = list(self.shape)
+            in_shape[0] *= 2
+            x = np.random.random(in_shape).astype(self.dtype)
+            y_1 = np.random.random(self.shape).astype(self.dtype)
+            y_2 = np.random.random(self.shape).astype(self.dtype)
+            tensor_x = paddle.to_tensor(x)
+            tensor_y = []
+            tensor_y.append(paddle.to_tensor(y_1))
+            tensor_y.append(paddle.to_tensor(y_2))
+            gather_result = paddle.concat(tensor_y)
+            if pg.rank() == root:
+                task = pg.gather(tensor_x, tensor_y[root], root, True)
+                task.wait()
+                assert np.array_equal(tensor_x, gather_result)
+            else:
+                task = pg.gather(tensor_x, tensor_y[1 - root], root, True)
+                task.wait()
+
+        test_gather(0)
+        test_gather(1)
```

Comment:
You assume that this ut would only run when world size is 2. If world size is not 2, this ut does not work.